src/ray/core_worker/core_worker.cc

```diff
@@ -238,6 +248,24 @@ void CoreWorkerProcess::RunTaskExecutionLoop() {
   instance_.reset();
 }
 
+void CoreWorkerProcess::RunStatsService() {
+  absl::MutexLock lock(&CoreWorkerProcess::stats_mutex_);
+  if (CoreWorkerProcess::stats_thread_) {
+    return;
+  }
+  // Initialize stats.
+  const ray::stats::TagsType global_tags = {{ray::stats::ComponentKey, "core_worker"},
+                                            {ray::stats::VersionKey, "0.9.0.dev0"}};
+
+  ray::stats::Init(global_tags, RayConfig::instance().metrics_agent_port(),
+                   CoreWorkerProcess::stats_io_service_);
+  CoreWorkerProcess::stats_thread_.reset(
+      new std::thread([]() { CoreWorkerProcess::stats_io_service_.run(); }));
```

Comment:
We should stop this when the core worker is GC'ed

Reply:
> We should stop this when the core worker is GC'ed

@kfstorm told me that multiple core worker will be binded to one CoreWorkerProscess, so we'd better initilize stats module once and join the separated thread in the end of CoreWorkerProccess. I'll figure out how to deconstrut stats io service, io worker and stats thread safely.

Reply:
I think what @rkooo567 means is that we should stop the stats thread when `CoreWorkerProcess` is shutting down. @ashione If we don't have a way to stop the stats thread currently, could you please add a comment here to state the fact and reason?